zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/OriginalDataAgent.scala

```diff
@@ -12,31 +13,18 @@ import scala.concurrent.ExecutionContext
 
 class OriginalDataAgent(override val dbName: String,
                         override val schema: Schema,
+                        val initCardinality: Cardinality,
```

Comment:
I would suggest just pass a `DataSetInfo` or `Stats` to the agent which makes more sense. And we can put more logic here in the agent about how to use the `info` or `stats`.